### PR TITLE
TOOLS vagrant tweaks and increase verbosity on key init script/s 

### PR DESCRIPTION
**Pull request by davidbuzz**

### CODE REVIEW

File: Tools/scripts/build-jsbsim.sh [hunk 1]
```diff
@@ -2,9 +2,20 @@
 
 set -e
 set -x
+echo "---------- build-jsbsim.sh start ----------"
```
Comment (peterbarker) [inline Tools/scripts/build-jsbsim.sh]: Maybe just use $0 here

File: Tools/vagrant/initvagrant.sh [hunk 2]
```diff
@@ -1,24 +1,36 @@
...
+    echo "SORRY, MUST RUN THIS SCRIPT AS ROOT, GIVING UP"
+    exit 1
+fi
 
 VAGRANT_USER=ubuntu
 if [ -e /home/vagrant ]; then
     # prefer vagrant user
     VAGRANT_USER=vagrant
 fi
+echo USING VAGRANT_USER:$VAGRANT_USER
```
Comment (peterbarker) [inline Tools/vagrant/initvagrant.sh]: Isn't a lot of this coming out via `set -x` anyway?  Next thing people will want to remove `-x` again :-/

File: Vagrantfile [hunk 3]
```diff
@@ -38,39 +38,68 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
-    config.vm.provision :shell, path: "Tools/vagrant/initvagrant.sh"
+    config.vm.provider "virtualbox" do |vb|
+        vb.name = "ArduPilot (Xenial32)"
+        vb.gui = true
+    end
+    
   end
 
-  # EOL July 2018
+  # NO LONGER AVAILABLE FOR DOWNLOAD, EOL January 2018
```
Comment (peterbarker) [inline Vagrantfile]: Some of us still have the relevant boxes, however, which is why I haven't removed these.

File: Tools/vagrant/initvagrant.sh (hunk 2 above)
Comment (davidbuzz) [reply Tools/vagrant/initvagrant.sh]: In my setup at least, an explicit 'echo' actually puts the the output in green text, while the rest of the verbose text goes past in red, so the echo stands out.     I had this especially stand out like this, as the choice to use /home/vagrant over /home/ubuntu is important for a bunch of scripts later on ( such as where jsbsim actually gets installed ).

File: Tools/scripts/install-prereqs-ubuntu.sh [hunk 4]
```diff
@@ -134,4 +134,4 @@ apt-cache search arm-none-eabi
  git submodule init
  git submodule update
 )
-echo "---------- install-prereqs-ubuntu.sh end ----------"
\ No newline at end of file
+echo "---------- $0 end ----------"
```
Comment (khancyr) [inline Tools/scripts/install-prereqs-ubuntu.sh]: miss the new line at end of file

File: Tools/vagrant/initvagrant.sh [hunk 5]
```diff
@@ -1,24 +1,36 @@
...
+
 
 # artful rootfs is 2GB without resize:
 sudo resize2fs /dev/sda1
 
 usermod -a -G dialout $VAGRANT_USER
 
+echo "calling pre-reqs script..."
 /vagrant/Tools/scripts/install-prereqs-ubuntu.sh -y
+echo "...pre-reqs script done... initvagrant.sh continues."
```
Comment (OXINARF) [inline Tools/vagrant/initvagrant.sh]: These are the only ones I don't like as the command will already show up in the output.

**Comment by davidbuzz** [comment]
GUI mode in the vagrant file causes the "console" of the virtual box to be made visible while it's booting up and running, making it easier for new user/s to understand that there is a VM being used, that it's actually running,  and it allows them to also login to it without having to necessarily use SSH.  It does not impose the need for the linux environment to have X, or any other GUI, rather just giving easy access to a terminal.  it also makes it easier to turn off the VM, to see the disk activity of the VM , and to mess with other VM settings like how much ram is assigned to it.

**Comment by peterbarker** [comment]
On Thu, 5 Apr 2018, Buzz wrote:

> GUI mode in the vagrant file causes the "console" of the virtual box to be
> made visible while it's booting up and running, making it easier for new
> user/s to understand that there is a VM being used, that it's actually
> running, and it allows them to also login to it without having to
> necessarily use SSH. It does not impose the need for the linux environment
> to have X, or any other GUI, rather just giving easy access to a terminal.
> it also makes it easier to turn off the VM, to see the disk activity of
> the VM , and to mess with other VM settings like how much ram is assigned
> to it.

Yep, that's all true, but there's a couple of catches:
  - rebooting doesn't work well as the /vagrant mount stops working (IIRC)
  - it annoys the snot out of me when these pop up :-)

So - maybe we should change the main variants to actually install a full 
GUI environment and add some "-nongui" variants (e.g. "vagrant up 
trusty64-nogui")?

**Comment by peterbarker** [comment]
Other PR has been pushed in.

**Pull request closed without merging**